### ISSUE TITLE
list_hosts function does not return standalone hosts

### ISSUE THREAD

**Issue by derekmwright** [fog/fog-vsphere#101]
Might need to refactor this but I'm not sure what it will break. Here's what I'm thinking:

1. Default w/ no Arg is return all hosts within all datacenters, return `:datacenter` & `:cluster =|| nil` (nil indicating standalone) attributes
2. Provide `:datacenter` attribute to filter to DC, list all hosts across all clusters
3. Provide `:datacenter` & `:cluster` to filter hosts to a DC and cluster

**Comment by wyde19** [comment]
Yes, and if you have no any clusters in your deployment, all of this is effectively useless.
I will think of a solution in a moment.

**Comment by wyde19** [comment]
Okay, I have several questions.
Either I am dumb, or current logic is completely broken (in several places).
I had a hard time untangling this code, but that's what I got.

https://github.com/wyde19/fog-vsphere/commit/97e44675e84d859fe37c0e5e40eb2d3302dffb00
Please, make a review @jrgarcia and write what do you think about it.

Anyway, at least that really returns *hosts* right now.
Although, there are still space for improvements.
For example, I did not change `compute.hosts.all` yet, it still requires both `:datacenter` and `:cluster` (which is non-sense).

**Comment by wyde19** [comment]
@derekmwright I understand this issue is long due, but could you make few checks?
Please, check if instance of `ClusterComputeResource` when invoked with `#is_a?(ComputeResource)` returns `true`.
Also, do you have all those fields available on ComputeResource?
https://github.com/wyde19/fog-vsphere/commit/97e44675e84d859fe37c0e5e40eb2d3302dffb00#r27595863
